### DISCUSSION TITLE
ERROR: SQLITE_CANTOPEN: unable to open database file

### DISCUSSION THREAD

**Discussion by ccraig** [directus/directus#17823] [Archive]
I'm using the docker compose file from the Quickstart guide and it works great on my Mac (10.15) running Docker 20.10.10 and Compose 2.1.1

However, when I try running it on an up-to-date Ubuntu 22.04.2 LTS **VirtualBox host**, I receive this error:

` $ sudo docker compose up

[+] Running 1/0
 ⠿ Container directus-directus-1  Created                                                                                                                0.0s
Attaching to directus-directus-1
directus-directus-1  | [22:51:26.338] INFO: Initializing bootstrap...
directus-directus-1  | [22:51:51.366] ERROR: Can't connect to the database.
directus-directus-1  | [22:51:51.366] ERROR: SQLITE_CANTOPEN: unable to open database file
directus-directus-1  |     err: {
directus-directus-1  |       "type": "Error",
directus-directus-1  |       "message": "SQLITE_CANTOPEN: unable to open database file",
directus-directus-1  |       "stack":
directus-directus-1  |           Error: SQLITE_CANTOPEN: unable to open database file
directus-directus-1  |       "errno": 14,
directus-directus-1  |       "code": "SQLITE_CANTOPEN"
directus-directus-1  |     }
directus-directus-1 exited with code 1`

Docker version 20.10.21
Compose: 2.16.0

I remove the docker container, database/upload folders, and prune the volumes it creates before re-running. I even remove the image so it's a fresh pull.

I'm able to run other docker compose files without error. Not sure if it's a "sudo" thing since I don't have to use that when I run docker on my Mac.

**Comment by rowild** [comment]
In my case, I install Directus with a sudo-enabled user, but Directus creates everything on root. So I changed the owner of all those files after the first time this error occurs. 
`sudo chown $USER:$USER database`
Running "docker-compser up -d" again installs without error. But I have a feeling this will have follow-up problems... 

Check the comments and its links here:
https://github.com/directus/directus/discussions/15986

**Reply by ccraig** [reply]
That worked! However, I wonder as well if that will have follow-up problems.

**Reply by rowild** [reply]
Take a look at the links. It seems you that there is a way to sync client and docker ips...

**Reply by rowild** [reply]
I found this article to be helpful:
https://jtreminio.com/blog/running-docker-containers-as-current-host-user/

**Reply by ccraig** [reply]
That was interesting. I've done the uid/gid change trick in the past for other similar issues. Not my favorite route, but it was a great explanation.

**Comment by shawnweeks** [comment]
This is usually why you don't want to use directory mounts however there are ways to fix this in the container build but I can't find the source Dockerfile used for the Directus images anywhere. If someone points me at them I can probably suggest a PR to improve this behavior a bit. Another option would be to switch to volumes instead.

**Comment by glorinli** [comment]
Just chmod 777 for database and uploads then I can start it successfully.
